coursier.bzl

```diff
@@ -308,16 +308,44 @@ def _generate_coursier_command(repository_ctx):
         # The -noverify option seems to be required after the proguarding step
         # of the main JAR of coursier.
         java = repository_ctx.path(java_home + "/bin/java")
-        cmd = [java, "-noverify", "-jar", coursier]
+        cmd = [java, "-noverify", "-jar"] + _get_java_proxy_args(repository_ctx) + [coursier]
     elif repository_ctx.which("java") != None:
         # Use 'java' from $PATH
-        cmd = [repository_ctx.which("java"), "-noverify", "-jar", coursier]
+        cmd = [repository_ctx.which("java"), "-noverify", "-jar"] + _get_java_proxy_args(repository_ctx) + [coursier]
     else:
         # Try to execute coursier directly
-        cmd = [coursier]
+        cmd = [coursier] + _get_java_proxy_args(repository_ctx)
 
     return cmd
 
+# Extract the well-known environment variables http_proxy, https_proxy and
+# no_proxy and convert them to java.net-compatible property arguments.
+def _get_java_proxy_args(repository_ctx):
+    http_proxy = repository_ctx.os.environ.get("http_proxy") if repository_ctx.os.environ.get("http_proxy") else repository_ctx.os.environ.get("HTTP_PROXY")
```

Comment:
Does this also work with the capitalized versions of the environment variables? Is that common?

Reply:
I've never seen cases other than all lower- or all upper-case, although it's possible. Bazel only seems to implicitly work with the two forms I've checked here though: https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/ProxyHelper.java#L46

Reply:
I don't think there's special handling within a repository_ctx.os.env.get, is there? env.get is just a dictionary get. Feel free to skip this, we can add capitalized env var support when users request for it.

Reply:
There's at least a `default=` so we can shorten the assignment.